conans/test/conftest.py

```diff
@@ -73,11 +81,20 @@ def add_tool(request):
     env_tools = dict()
     for mark in request.node.iter_markers():
         if mark.name.startswith("tool_"):
+            version = mark.kwargs.get('version', 'default')
             tool_name = mark.name[5:]
-            if tool_name in tool_locations:
-                add_tools.append(tool_locations[tool_name])
-            if tool_name in tool_environments:
-                env_tools.update(tool_environments[tool_name])
+            if tool_name in tool_locations and tool_locations[tool_name].get(platform.system()):
```

Comment:
the code starts to become complicated and hard to read...
maybe if we extract `tool_locations[tool_name].get(platform.system())` (seems to be a repeating pattern) it will look better? or maybe split it.